runner/host/dmesg.go

```diff
@@ -41,6 +42,21 @@ func NewDMesgWithContext(ctx context.Context, cfg DMesgConfig) (*DMesg, error) {
 	if ctx == nil {
 		ctx = context.Background()
 	}
+	if runtime.GOOS == "darwin" {
+		shell, err := runner.NewShellWithContext(ctx, runner.ShellConfig{
+			Command:    "sudo dmesg",
```

Comment:
We generally require users to run hcdiag with permissions get run any of the commands it requires, so I would leave `sudo` out of this command and just make it `dmesg`.